### PR TITLE
feat: support animated (gif) picture frames

### PR DESCRIPTION
**Pull request by pbosio**
- added plugin to support decode gif binary.
- modified `NFTShapeLoaderController` class to load animated picture frames.

test scene: `-103.102.nft`

### CODE REVIEW

**Review by pravusjif — APPROVED** [review]

File: unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs [hunk 1]
```diff
@@ -266,9 +282,108 @@ void InitializePerlinNoise()
...
+    interface INFTAsset : IDisposable
+    {
+        Texture2D texture { get; }
+        int width { get; }
+        int height { get; }
+    }
+
+    class NFTImageAsset : INFTAsset
+    {
+        private Texture2D _texture;
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs]:
```suggestion
        public Texture2D texture { get; private set; }
```

File: unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs [hunk 2]
```diff
@@ -266,9 +282,108 @@ void InitializePerlinNoise()
...
+
+        public NFTImageAsset(Texture2D t)
+        {
+            _texture = t;
+        }
+    }
+
+    class NFTGifAsset : INFTAsset
+    {
+        DCLGif _gif;
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs]:
```suggestion
        DCLGif gif;
```

File: unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs [hunk 3]
```diff
@@ -266,9 +282,108 @@ void InitializePerlinNoise()
...
+        Coroutine _updateRoutine = null;
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs]:
```suggestion
        Coroutine updateRoutine = null;
```

File: unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs [hunk 4]
```diff
@@ -266,9 +282,108 @@ void InitializePerlinNoise()
...
+        public void SetUpdateTextureCallback(Action<Texture2D> callback)
+        {
+            _gif.OnFrameTextureChanged += callback;
+            _updateRoutine = CoroutineStarter.Start(_gif.UpdateRoutine());
+        }
+
+        public NFTGifAsset(DCLGif gif)
         {
-            UnityEngine.Object.Destroy(nftImageTexture);
+            _gif = gif;
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs]:
```suggestion
            this.gif = gif;
```

File: unity-client/Assets/UniGif/Example/Script/UniGifImage.cs [hunk 5]
```diff
@@ -0,0 +1,354 @@
+﻿/*
```
Comment (BrianAmadori) [inline unity-client/Assets/UniGif/Example/Script/UniGifImage.cs]: The example scripts have to be added to the project? We can't remove them?

File: unity-client/Assets/UniGif/Example/Script/Utility/UniGifImageAspectController.cs [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+UniGif
+Copyright (c) 2015 WestHillApps (Hironari Nishioka)
+This software is released under the MIT License.
+http://opensource.org/licenses/mit-license.php
+*/
+
+using UnityEngine;
+
+[ExecuteInEditMode]
+public class UniGifImageAspectController : MonoBehaviour
```
Comment (BrianAmadori) [inline unity-client/Assets/UniGif/Example/Script/Utility/UniGifImageAspectController.cs]: Can we move the `UniGif` folder to `Plugins`?

File: unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs [hunk 7]
```diff
@@ -266,9 +282,108 @@ void InitializePerlinNoise()
...
+            if (_gif != null)
+            {
+                _gif.Dispose();
+            }
+        }
+
+        public void SetUpdateTextureCallback(Action<Texture2D> callback)
+        {
+            _gif.OnFrameTextureChanged += callback;
+            _updateRoutine = CoroutineStarter.Start(_gif.UpdateRoutine());
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs]: We shouldn't be stopping this coroutine if its already running?

File: unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs [hunk 8]
```diff
@@ -266,9 +282,108 @@ void InitializePerlinNoise()
...
+            }
+        }
+
+        public NFTImageAsset(Texture2D t)
+        {
+            _texture = t;
+        }
+    }
+
+    class NFTGifAsset : INFTAsset
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTShapeLoaderController.cs]: Why are we using the `NFT` moniker here? This means we can't use this code for a non-NFT gif component? If we can, we should choose a more generic name.

**Review by BrianAmadori — CHANGES_REQUESTED** [review]

**Review by BrianAmadori — APPROVED** [review]

**Pull request merged**